libs/MotionKit/source/MotionKit.cpp

```diff
@@ -58,61 +54,74 @@ void MotionKit::startStabilisation()
 {
 	stop();
 
-	_imukit.start();
-	_imukit.setOrigin();
+	_euler_angles_previous = _imukit.getEulerAngles();
+	_pid.setTargetYaw(_euler_angles_previous.yaw);
 
 	_target_not_reached		  = false;
 	_stabilisation_requested  = true;
 	_rotate_x_turns_requested = false;
 
-	_event_loop.start();
+	auto on_euler_angles_rdy_callback = [this](const interface::EulerAngles &euler_angles) {
+		iterateOverRotation(euler_angles);
```

Comment:
it's quite complicated to understand why it is sometimes called with `direction` and sometimes without.

it would be better to have two different functions to make it clearer for the user/reader